tests/Functional/Action/WebHook/UpdateLineItemsActionTest.php

```diff
@@ -0,0 +1,365 @@
+                            'eventData' => [
+                                'alias' => 'qti-interactions-delivery',
+                            ],
+                        ],
+                    ],
+                ],
+                'expectedMessage' => $missingDeliveryUri,
+            ],
+            'aliasWrongType' => [
+                'requestBody' => [
+                    'source' => 'https://someinstance.taocloud.org/',
+                    'events' => [
+                        [
+                            'eventId' => '52a3de8dd0f270fd193f9f4bff05232f',
+                            'triggeredTimestamp' => 1565602371,
+                            'eventData' => [
+                                'alias' => 123,
+                                'deliveryURI' => 'https://docker.localhost/ontologies/tao.rdf#FFF',
+                            ],
+                        ],
+                    ],
+                ],
+                'expectedMessage' => $invalidAliasTypeMessage,
+            ],
+        ];
+    }
+
+    private function getSuccessRequestBody(): array
+    {
+        return [
+            'source' => 'https://someinstance.taocloud.org/',
+            'events' => [
+                [
+                    'eventId' => '52a3de8dd0f270fd193f9f4bff05232f',
+                    'eventName' => 'WrongEvent',
+                    'triggeredTimestamp' => 1565602371,
+                    'eventData' => [
+                        'alias' => 'qti-interactions-delivery',
+                        'deliveryURI' => 'https://docker.localhost/ontologies/tao.rdf#FFF',
+                    ],
+                ],
+                [
+                    'eventId' => '52a3de8dd0f270fd193f9f4bff05232f',
+                    'eventName' => 'RemoteDeliveryPublicationFinished',
+                    'triggeredTimestamp' => 1565602371,
+                    'eventData' => [
+                        'alias' => 'wrong-alias',
+                        'deliveryURI' => 'https://docker.localhost/ontologies/tao.rdf#FFF',
+                    ],
+                ],
+                [
+                    'eventId' => '52a3de8dd0f270fd193f9f4bff05232c',
+                    'eventName' => 'RemoteDeliveryPublicationFinished',
+                    'triggeredTimestamp' => 1565602380,
+                    'eventData' => [
+                        'alias' => 'lineItemSlug',
+                        'deliveryURI' => 'https://docker.localhost/ontologies/tao.rdf#FFF',
+                    ],
+                ],
+                [
+                    'eventId' => '52a3de8dd0f270fd193f9f4bff05232c',
+                    'eventName' => 'RemoteDeliveryPublicationFinished',
+                    'triggeredTimestamp' => 1565602390,
+                    'eventData' => [
+                        'alias' => 'lineItemSlug',
+                        'deliveryURI' => 'https://docker.localhost/ontologies/tao.rdf#RightOne',
+                    ],
```

Comment:
👍  nice, but you could play also with the order of events to see if you properly filter by last triggered at date